pkg/fanal/image/image.go

```diff
@@ -123,6 +124,85 @@ func GuessBaseImageIndex(histories []v1.History) int {
 			foundNonEmpty = true
 		}
 
+		// Hack to handle golang base images which don't have a CMD/Entrypoint instruction
+		// i != len(histories)-1 : is to handle scenarios where the golang base image itself is being scanned
+		/*
+			Go version 1.16-latest tend to have the following instructions in the last 5 layers
+
+			ENV GOPATH /go
+			ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
+			COPY --from=build --link /usr/local/go/ /usr/local/go/  ............ [not present in older versions]
+			RUN mkdir -p "$GOPATH/src" "$GOPATH/bin" && chmod -R 1777 "$GOPATH"
+			WORKDIR $GOPATH
+		*/
+		// We are looking for an occurence of the same
+		if i != len(histories)-1 && (strings.HasPrefix(h.CreatedBy, "/bin/sh -c #(nop) WORKDIR ") ||
+			strings.HasPrefix(h.CreatedBy, "WORKDIR ")) {
+
+			// check if we have a subset of 5 layers available since WORKDIR is encountered
+			if (i-4) >= 0 && i+1 < len(histories) {
```

Comment:
Not sure if it makes it readable.

We are iterating over the image history in a reverse order, so want to be explicit that we are looking at a layer index (i-4) and same is being used while extracting the slice 
`for _, cmd := range histories[i-4 : i+1] `

Intended to keep it consistent.

Reply:
The condition is that the i'th layer we are iterating is in the 4 to last layer, correct. So is was going with that perspective
Also do we need the i+1 < len(histories) condition since this condition will always hold true due to line 119 `for i := len(histories) - 1; i >= 0; i--`

Reply:
Yeah you are right its redundant. Leaving it as a precautionary out of bounds check